File: str/reverseVowls.java
public String reverse(String input) {
    // Write your solution here
    if(input == null || input.length <= 1){
      return input;
    }
    Set<Character> set = new HashSet<>();
    set.add('a');
    set.add('e');
    set.add('i');
    set.add('o');
    set.add('u');

    char[] arr = input.toCharArray();
    int left = 0;
    int right = arr.length-1;

    while(left < right){
      if(!set.contains(arr[left])){
        left++;
        continue;
      }
      if(!set.contains(arr[right])){
        right--;
        continue;
      }
      Character temp = arr[left];
      arr[left] = arr[right];
      arr[right] = temp;
      left++;
      right--;
    }
    return new String(arr);
}

File: str/reverse/reverseVowls.py
def reverseVowls(input):
    if input is not None or not input:
        return input
    visited = set('a', 'e', 'i', 'o', 'u')
    left = 0
    right = len(input) - 1
    arr = list(input)

    while left < right:
        if arr[left] not in visited:
            left += 1
            continue
        if arr[right] not in visited:
            right -= 1
            continue
        arr[left], arr[right] = arr[right], arr[left]
        left += 1
        right -= 1

    return ''.join(arr)
